R/08-create-final-reports.R

```diff
@@ -0,0 +1,429 @@
+  ) %>% 
+  mutate(
+    GRADS = GRADUATES*PERCENT
+  )
+
+# calculate totals by year and age group 
+grads_agg <- filtered_grads %>% 
+  mutate(PSSM_CREDENTIAL = toupper(PSSM_CREDENTIAL)) %>% # include so things aggregate correctly 
+  inner_join(
+    age_groups ,
+    by = c('AGE_GROUP' = 'AGE_GROUP_LABEL')
+  ) %>% 
+  inner_join(
+    age_groups_rollup,
+    by = 'AGE_GROUP_ROLLUP'
+  ) %>% 
+  group_by(
+    AGE_GROUP_ROLLUP_LABEL,
+    YEAR,
+    PSSM_CREDENTIAL
+  ) %>% 
+  summarize(GRADS = sum(GRADS)) %>% 
+  inner_join(credentials %>% mutate(PSSM_CREDENTIAL = toupper(PSSM_CREDENTIAL)), by='PSSM_CREDENTIAL') %>% 
+  select(
+    AGE_GROUP_ROLLUP_LABEL,
+    YEAR,
+    PSSM_CREDENTIAL_NAME,
+    GRADS
+  ) %>%
+  ungroup()
+
+# round each value to nearest 5
+grads_rounded <- grads_agg %>% 
+  mutate(GRADS = as.integer(5*round(GRADS/5, 0)))
+
+# get totals per age_group and join back in 
+grad_totals <- grads_rounded %>% 
+  group_by(AGE_GROUP_ROLLUP_LABEL, YEAR) %>% 
+  summarize(GRADS = sum(GRADS), PSSM_CREDENTIAL_NAME = 'Total')
+grad_totals <- grads_agg %>% 
+  group_by(AGE_GROUP_ROLLUP_LABEL, YEAR) %>% 
+  summarize(GRADS = sum(GRADS), PSSM_CREDENTIAL_NAME = 'Total')  %>% 
+  mutate(GRADS = as.integer(5*round(GRADS/5, 0)))
+# pivot years to columns, sort, and make nice names
+grads <- grads_rounded %>% 
+  bind_rows(grad_totals) %>% 
+  pivot_wider(names_from = YEAR, values_from = GRADS) %>% 
+  mutate(is_total = PSSM_CREDENTIAL_NAME == 'Total') %>% 
+  arrange(AGE_GROUP_ROLLUP_LABEL, is_total, PSSM_CREDENTIAL_NAME) %>% 
+  rename(
+    `Age Group` = AGE_GROUP_ROLLUP_LABEL,
+    `Credential Type` = PSSM_CREDENTIAL_NAME
+    ) %>% 
+  select(-is_total)
+
+grads
+
+# 2. Create Final Occupation Projections ---- 
+
+# join all 3 models together and create 
+# QI (quality indicator) and 
+# CI (coverage indicator) columns
+years <- tmp_tbl_model %>% select(starts_with('X')) %>% names()
+# try to do this generically without mention of first year 
+first_year_col <- years[order(years)][1]
+
+tmp_occ <- tmp_tbl_model %>% 
+  mutate(first_year = .[first_year_col] %>% pull()) %>% 
+  left_join(
+    tmp_tbl_qi %>% 
+      mutate(QI=.[first_year_col] %>% pull()) %>% 
+      select(Expr1,QI),
+    by="Expr1"
+    ) %>% 
+  left_join(
+    tmp_tbl_ptib %>% 
+      mutate(CI=.[first_year_col] %>% pull()) %>% 
+      select(Expr1,CI),
+    by="Expr1"
+    ) %>% 
+  filter(NOC_Level==5) %>% 
+  arrange(
+    Age_Group_Rollup_Label,
+    NOC,
+    Current_Region_PSSM_Code_Rollup
+    ) %>% 
+  mutate(`Public Post-Secondary Coverage Indicator`=ifelse(is.na(CI),0,first_year/CI)) %>% 
+  mutate(QI_calc = (abs(first_year-QI)/QI)) %>% 
+  mutate(`Quality Indicator`= case_when(QI_calc < 0.25 ~ QI_calc,
+                                        (first_year < 10 | QI < 10 | is.na(first_year) | is.na(QI)) ~ NA_integer_,
+                                        TRUE ~ QI_calc)) %>% 
+  # round outputs 
+  mutate(across(starts_with('X'), ~ceiling(.)))
```

Comment:
I think we need to round to nearest whole number, not up.
```suggestion
  mutate(across(starts_with('X'), ~round(.)))
```